### PR TITLE
Enable rqt_console in ROS2 Dashing-Diademata

### PR DESCRIPTION
**Pull request by vinnamkim**
Signed-off-by: vinnamkim <vinnam.kim@gmail.com>

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please provide a description of the patch and why certain parts are removed.

**Comment by vinnamkim** [comment]
@dirk-thomas 

Sorry for my lack of information. I found that rqt_console is not working when testing ROS2(latest Dashing) application of my works. I tried to make minimal changes to show log messages in rqt_console. Tested this commit with [demo_nodes_cpp/talker](https://github.com/ros2/demos/blob/86f16deb442932304ea2944d4a9d80d21db73e48/demo_nodes_cpp/src/topics/talker.cpp#L56) and [demo_nodes_py/talker](https://github.com/ros2/demos/blob/86f16deb442932304ea2944d4a9d80d21db73e48/demo_nodes_py/demo_nodes_py/topics/talker.py#L34)

File: src/rqt_console/console.py [hunk 1]
```diff
@@ -100,7 +100,6 @@ def convert_rosgraph_log_message(log_msg):
         msg.message = log_msg.msg
         msg.severity = log_msg.level
         msg.node = log_msg.name
-        msg.stamp = (log_msg.header.stamp.sec, log_msg.header.stamp.nanosec)
```
Comment (vinnamkim) [inline src/rqt_console/console.py:103]: `log_msg` obtained from `/rosout` doesn't have `header` entity. So, I decided to remove this.

File: src/rqt_console/message_data_model.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
 class MessageDataModel(QAbstractTableModel):
 
     # the column names must match the message attributes
-    columns = ['message', 'severity', 'node', 'stamp', 'topics', 'location']
```
Comment (vinnamkim) [inline src/rqt_console/message_data_model.py:43]: Unlike ROS1, It seems ROS2 log messages don't have `topics` entity, so I dropped it.

**Comment by hidmic** [comment]
@vinnamkim I took the liberty to push a few more fixes to get this one going. Also, I think we should probably merge this onto a `dashing-devel` branch as AFAIK these changes are not Crystal compatible. @dirk-thomas PTAL.

**Comment by dirk-thomas** [comment]
Retargeted the `dashing-devel` branch and updated the branches in ros/rosdistro#22496 as well as https://github.com/ros2-gbp/rqt_console-release/commit/55b1e88c88f610859119b6432018d0256501e458.

File: src/rqt_console/console.py [hunk 3]
```diff
@@ -142,5 +141,8 @@ def trigger_configuration(self):
     def _subscribe(self, topic):
         if self._subscriber:
             self._context.node.destroy_subscription(self._subscriber)
-        self._subscriber = self._context.node.create_subscription(Log, topic, self.queue_message)
+        from rclpy.qos import qos_profile_system_default
```
Comment (dirk-thomas) [inline src/rqt_console/console.py]: If there is no reason why this import is local please move it to the top of the file.

Comment (hidmic) [reply src/rqt_console/console.py]: A code reflex, to not import modules that cause C extension loading at the module level. It doesn't make sense really. Fixed in https://github.com/ros-visualization/rqt_console/pull/10/commits/ace4a4fb90cac203a2e0f31bf5ac17bbb20800a9.

**Comment by dirk-thomas** [comment]
Thank you both for this patch.

**Pull request merged**